core/ipc/sink/aws-sqs-impl/src/main/java/org/opennms/core/ipc/sink/aws/sqs/AwsRemoteMessageDispatcherFactory.java

```diff
@@ -0,0 +1,199 @@
+
+package org.opennms.core.ipc.sink.aws.sqs;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.jms.JMSException;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import org.opennms.core.ipc.sink.api.Message;
+import org.opennms.core.ipc.sink.api.MessageConsumerManager;
+import org.opennms.core.ipc.sink.api.SinkModule;
+import org.opennms.core.ipc.sink.common.AbstractMessageDispatcherFactory;
+import org.opennms.core.logging.Logging;
+import org.opennms.core.logging.Logging.MDCCloseable;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazon.sqs.javamessaging.SQSConnection;
+import com.codahale.metrics.JmxReporter;
+
+/**
+ * A factory for creating AwsRemoteMessageDispatcher objects.
+ * 
+ * @author <a href="mailto:agalue@opennms.org">Alejandro Galue</a>
+ */
+public class AwsRemoteMessageDispatcherFactory extends AbstractMessageDispatcherFactory<String> {
+
+    /** The Constant LOG. */
+    private static final Logger LOG = LoggerFactory.getLogger(AwsRemoteMessageDispatcherFactory.class);
+
+    /** The AWS configuration. */
+    private final Properties awsConfig = new Properties();
+
+    /** The configuration administration object. */
+    private ConfigurationAdmin configAdmin;
+
+    /** The reporter. */
+    private JmxReporter reporter;
+
+    /** The SQS session. */
+    private Session session;
+
+    /** The SQS Connection. */
+    private SQSConnection connection;
+
+    /** The producers. */
+    private Map<String,MessageProducer> producers = new HashMap<>();
+
+    /* (non-Javadoc)
+     * @see org.opennms.core.ipc.sink.common.AbstractMessageDispatcherFactory#getModuleMetadata(org.opennms.core.ipc.sink.api.SinkModule)
+     */
+    @Override
+    public <S extends Message, T extends Message> String getModuleMetadata(final SinkModule<S, T> module) {
+        return AwsUtils.getQueueName(module);
+    }
+
+    /* (non-Javadoc)
+     * @see org.opennms.core.ipc.sink.common.AbstractMessageDispatcherFactory#dispatch(org.opennms.core.ipc.sink.api.SinkModule, java.lang.Object, org.opennms.core.ipc.sink.api.Message)
+     */
+    @Override
+    public <S extends Message, T extends Message> void dispatch(SinkModule<S, T> module, String topic, T message) {
+        try (MDCCloseable mdc = Logging.withPrefixCloseable(MessageConsumerManager.LOG_PREFIX)) {
+            LOG.trace("dispatch({}): sending message {}", topic, message);
+
+            try {
+                TextMessage msg = session.createTextMessage(module.marshal((T)message));
+                getProducer(getModuleMetadata(module)).send(msg);
+            } catch (JMSException e) {
+                LOG.error("Error occured while sending message to topic {}.", topic, e);
```

Comment:
If the Minion has no internet connectivity, I suspect we'll end up hitting this block and dropping all of the messages.

Reply:
You're right! Internet connectivity is mandatory is you want to do something with Amazon Web Services outside a VPC. That being said, how do you suggest to do to handle this fact ?